src/coreclr/tools/Common/JitInterface/CorInfoImpl.cs

```diff
@@ -833,6 +833,11 @@ private void Get_CORINFO_SIG_INFO(MethodDesc method, CORINFO_SIG_INFO* sig, Meth
         {
             Get_CORINFO_SIG_INFO(method.Signature, sig, scope);
 
+            // Get_CORINFO_SIG_INFO above will have correctly set CORINFO_CALLCONV_ASYNCCALL for async variants,
```

Comment:
Why special case these methods here instead of just in the Get_CORINIFO_SIGN_INFO? We also could create wrapper MethodDescs for the void-returning async intrinsics like we do / plan to do for async variants.